debugger/src/debug_adapter.rs

```diff
@@ -1125,41 +1157,219 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
+                            // If we find one, we use it ONLY if it is different from the previous one (stored_source_location).
+                            // - This helps to reduce visual noise in the VSCode UX, by not displaying the same line of source code multiple times over.
+                            // If we do not find a source location, then just return the raw assembly without file/line/column information.
+                            let mut location = None;
+                            let mut line = None;
+                            let mut column = None;
+                            if let Some(current_source_location) = core_data
+                                .debug_info
+                                .get_source_location(instruction.address()) {
+                                if let Some(previous_source_location) = stored_source_location.clone() {
+                                    if current_source_location != previous_source_location {
+                                        location = get_dap_source(&current_source_location);
+                                        line = current_source_location.line.map(|line| line as i64);
+                                        column = current_source_location.column.map(|col| match col {
+                                            ColumnType::LeftEdge => 0_i64,
+                                            ColumnType::Column(c) => c as i64,
+                                        });
+                                        stored_source_location = Some(current_source_location);
+                                    }
+                                } else {
+                                        stored_source_location = Some(current_source_location);
+                                }
+                            } else {
+                                // It won't affect the outcome, but log it for completeness.
+                                log::debug!("The request `Disassemble` could not resolve a source location for memory reference: {:#010}", instruction.address());
+                            }
+
+                            // Create the instruction data.
+                            dap_types::DisassembledInstruction {
+                                address: format!("{:#010X}", instruction.address()),
+                                column,
+                                end_column: None,
+                                end_line: None,
+                                instruction: format!(
+                                    "{}  {}",
+                                    instruction.mnemonic().unwrap_or("<unknown>"),
+                                    instruction.op_str().unwrap_or("")
+                                ),
+                                instruction_bytes: None,
+                                // Use the code below to include machine code. My personal opinion is that it adds too much noise to an already noisy UX.
```

Comment:
Dead code :)

Reply:
Yes, I mostly left it here, in case some diligent reviewer asks why I didn't provide the machine code :)